### PR TITLE
Use the AMO ID for reviewer links

### LINKED ISSUE #10826: Reviewer links on listing page should use the AMO ID instead of slug
### Describe the problem and steps to reproduce it:

The listing page includes links to reviewer tools (if you have permission).

### What happened?

The links use the add-on slug.

### What did you expect to happen?

The links use the AMO ID.

### Anything else we should know?

The AMO ID became the canonical identifier in https://github.com/mozilla/addons-server/commit/8c81b12efe5181b233ddacb4442c6f7ac9d95256. This update here saves us a redirect.

### PR DESCRIPTION
**Pull request by wagnerand**
Fix #10826.

### CODE REVIEW

File: tests/unit/amo/components/TestAddonAdminLinks.js [hunk 1]
```diff
@@ -22,6 +22,7 @@ import {
 
 describe(__filename, () => {
   let store;
+  const id = 123;
```
Comment (willdurand) [inline tests/unit/amo/components/TestAddonAdminLinks.js:25]: Please rename this variable to `addonId` if you intend to use it in test cases below.

File: tests/unit/amo/components/TestAddonAdminLinks.js [hunk 2]
```diff
@@ -22,6 +22,7 @@ import {
...
   const slug = 'some-slug';
```
Comment (willdurand) [inline tests/unit/amo/components/TestAddonAdminLinks.js:26]: Is the `slug` still used in this test file?

**Review by willdurand — COMMENTED** [review]
I had some minor comments

File: tests/unit/amo/components/TestAddonAdminLinks.js (hunk 2 above)
Comment (wagnerand) [reply tests/unit/amo/components/TestAddonAdminLinks.js:26]: Yes, for the link to devhub.

**Review by diox — APPROVED** [review]

**Pull request merged**